### PR TITLE
Add WebP support to image decoder

### PR DESCRIPTION
**Pull request by lbhm**
## Description
- [ ] **Bug fix** (*non-breaking change which fixes an issue*)
- [x] **New feature** (*non-breaking change which adds functionality*)
- [ ] **Breaking change** (*fix or feature that would cause existing functionality to not work as expected*)
- [ ] **Refactoring** (*Redesign of existing code that doesn't affect functionality*)
- [ ] **Other** (*e.g. Documentation, Tests, Configuration*)

### What happened in this PR

This PR adds WebP support to the image decoder as suggested in  #3203. WebP decoding is performed using OpenCV.
**Edit:** I think I kinda messed up this PR by first merging a change from this repo into my fork. Is this a problem?

#### Additional information
Affected modules and functionalities:
 - `dali/image`: Added class for WebP images and extended `image_factory.cc`
 - `dali/operators`: Added webp to `reader/loader/utils.h`

Key points relevant for the review:

- The most critical part IMO is to verify `WebpImage::PeekShapeImpl` in `webp.cc`. I gave the format references I used to decode shape information as comments. As noted in the comments, the sync code for lossy files is not where I would expect it based on my understanding of the format reference but all my tests ran through.
- Please advise if I extended the docs in the correct location(s).
- Please advise if I modified the test correctly.
- As I am a C++ novice, I welcome any feedback on coding style, inefficiencies, etc.

### Tests
- [ ] Existing tests apply
- [x] New tests added
  - [x] Python tests
  - [ ] GTests
  - [ ] Benchmark
  - [ ] Other
- [ ] N/A

### Documentation
- [ ] Existing documentation applies
- [x] Documentation updated
  - [ ] Docstring
  - [ ] Doxygen
  - [ ] RST
  - [ ] Jupyter
  - [ ] Other
- [ ] N/A

#### Requirements
- [ ] Implements new requirements
- [ ] Affects existing requirements
- [ ] N/A

**REQ IDs**: N/A

**JIRA TASK**: N/A

### CODE REVIEW

File: dali/operators/decoder/image_decoder.cc [hunk 1]
```diff
@@ -185,7 +190,7 @@ Please note that GPU acceleration for JPEG 2000 decoding is only available for C
 // Fused
 
 DALI_SCHEMA(decoders__ImageCrop)
-  .DocStr(R"code(Decodes images and extracts regions-of-interest (ROI) that are specified
+.DocStr(R"code(Decodes images and extracts regions-of-interest (ROI) that are specified
```
Comment (JanuszL) [inline dali/operators/decoder/image_decoder.cc:193]: Please restore the indentation.

File: dali/test/python/test_utils_tensorflow.py [hunk 2]
```diff
@@ -246,6 +246,7 @@ def compare(dataset_results, standalone_results, iterations=-1,  num_devices=1):
 
 def run_tf_dataset_graph(device, device_id=0, get_pipeline_desc=get_image_pipeline,
         to_dataset=to_image_dataset, to_stop_iter=False):
+    tf.compat.v1.reset_default_graph()
```
Comment (JanuszL) [inline dali/test/python/test_utils_tensorflow.py:249]: Could you rebase on top of the most recent main branch?

File: qa/TL1_jupyter_plugins/test_pytorch.sh [hunk 3]
```diff
@@ -1,7 +1,9 @@
 #!/bin/bash -e
 
 # used pip packages
-pip_packages="pillow jupyter matplotlib torchvision torch pytorch-lightning"
+# nvidia-index provides a stub for tensorboard which collides with one required by pytorch-lightning
```
Comment (JanuszL) [inline qa/TL1_jupyter_plugins/test_pytorch.sh:4]: Same as test_utils_tensorflow.py.

**Comment by JanuszL** [comment]
@lbhm - thank you for your contribution. I have only a handful of nitpicks.

**Comment by lbhm** [comment]
@JanuszL: I think my superficial git knowledge caused me messing up my local DALI branch more than before. I ended up creating a new branch based on the current main branch and cherrypicked my commits into that branch. Since I cannot change the source branch of a PR on GitHub, I'll recreate this PR and delete the old (this) one once the new one has been accepted.

**Comment by JanuszL** [comment]
> @JanuszL: I think my superficial git knowledge caused me messing up my local DALI branch more than before. I ended up creating a new branch based on the current main branch and cherrypicked my commits into that branch. Since I cannot change the source branch of a PR on GitHub, I'll recreate this PR and delete the old (this) one once the new one has been accepted.

I'm fine with that. Whatever works for you.

**Pull request closed without merging**